src/Storage/Services/DataService.cs

```diff
@@ -45,5 +50,41 @@ public async Task StartFileScan(Instance instance, DataType dataType, DataElemen
                 await _fileScanQueueClient.EnqueueFileScan(serialisedRequest, ct);
             }
         }
+
+        /// <inheritdoc/>
+        public async Task<(string String, ServiceError ServiceError)> GenerateSha256Hash(string org, Guid instanceGuid, Guid dataElementId)
+        {
+            DataElement dataElement = await _dataRepository.Read(instanceGuid, dataElementId);
+            if (dataElement == null)
+            {
+                return (null, new ServiceError(404, $"DataElement not found, dataElementId: {dataElementId}"));
+            }
+
+            Stream filestream = await _dataRepository.ReadDataFromStorage(org, dataElement.BlobStoragePath);
+            if (filestream == null || filestream.Length == 0)
+            {
+                return (null, new ServiceError(404, $"Failed reading file, dataElementId: {dataElementId}"));
+            }
+
+            return (CalculateSha256Hash(filestream), null);
```

Comment:
are we sure errors cannot occur during the hashing? E.g. with empty files?

Reply:
Filestream is checked not to be null or empty before this point. Any other errors should be 500 from controller.